tools/cordon-cluster/main/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+var (
+	scheme         = runtime.NewScheme()
+	ctx            = context.Background()
+	kubeconfigPath = os.Getenv("KUBECONFIG")
+)
+
+func main() {
+	hubClusterContext := flag.String("hubClusterContext", "", "the kubectl context for the hub cluster")
+	clusterName := flag.String("clusterName", "", "name of the cluster to cordon")
+	flag.Parse()
+
+	if *hubClusterContext == "" {
+		log.Fatalf("hub cluster context for kubectl cannot be empty")
+	}
+	if *clusterName == "" {
+		log.Fatalf("cluster name to cordon cannot be empty")
+	}
+
+	if err := clusterv1beta1.AddToScheme(scheme); err != nil {
+		log.Fatalf("failed to add custom APIs (cluster) to the runtime scheme: %v", err)
+	}
+	if err := placementv1beta1.AddToScheme(scheme); err != nil {
+		log.Fatalf("failed to add custom APIs (placement) to the runtime scheme: %v", err)
+	}
+
+	clusterConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
what is this for?

Reply:
clusterConfig -> restConfig -> hubClient, similar setup to our E2E